tests/conftest.py

```diff
@@ -502,3 +503,15 @@ def isolated_fs(fs):
     # add cassettes dir
     cassettes_dir = join(dirname(realpath(__file__)), "cassettes")
     fs.add_real_directory(cassettes_dir)
+
+
+def assert_invoke_exited_with(result: Result, exit_code: int):
+    msg = f"Expected code {exit_code}, got {result.exit_code}"
+    if result.exit_code != exit_code:
+        sys.stderr.write(msg)
+        sys.stdout.write(msg)
+    assert result.exit_code == exit_code, msg
```

Comment:
Oh, I misunderstood your screenshot. This is not what I had in mind.

Originally, tests running ggshield commands would do so like this:

```python
result = cli_fs_runner.invoke(cli, cmd)
assert result.exit_code == 0
```

The problem with this is that if the command fails, we don't know what it printed, because the stdout and the stderr of the command is in `result.output`. This makes the failure hard to debug.

To work around that, we started replacing the `assert` call with this:

```python
assert result.exit_code == 0, result.output
```

This way, if a command fails pytest prints what the failed command printed on stdout.

The goal of this new `assert_invoke_exited_with()` function is to systematize this idea of printing what the command printed in case of errors. To do so you must add to `msg` the content of `result.stdout` and `result.stderr` (for the cases where stderr is not merged into `result.stderr`)

Reply:
Thanks for @agateau-gg for the clarification. I pushed some changes that I believe correspond to what you were expecting, tell me if that's not the case!